src/backend_model_instance.cc

```diff
@@ -193,135 +186,62 @@ TritonModelInstance::~TritonModelInstance()
 }
 
 Status
-TritonModelInstance::SetInstances(
-    TritonModel* model,
-    const triton::common::BackendCmdlineConfigMap& backend_cmdline_config_map,
-    const triton::common::HostPolicyCmdlineConfigMap& host_policy_map,
-    const inference::ModelConfig& model_config)
+TritonModelInstance::CreateInstance(
+    TritonModel* model, const std::string& name, const Signature& signature,
+    TRITONSERVER_InstanceGroupKind kind, int32_t device_id,
+    const std::vector<std::string>& profile_names, const bool passive,
+    const std::string& host_policy_name,
+    const inference::ModelRateLimiter& rate_limiter_config,
+    const std::vector<SecondaryDevice>& secondary_devices,
+    std::shared_ptr<TritonModelInstance>* triton_model_instance)
 {
-  std::unordered_map<
-      Signature, std::vector<std::shared_ptr<TritonModelInstance>>>
-      existing_instances = model->IndexInstances();
-
   static triton::common::HostPolicyCmdlineConfig empty_host_policy;
+  const triton::common::HostPolicyCmdlineConfig* host_policy;
+  const auto policy_it = model->HostPolicyMap().find(host_policy_name);
+  if (policy_it != model->HostPolicyMap().end()) {
+    host_policy = &policy_it->second;
+  } else {
+    host_policy = &empty_host_policy;
```

Comment:
Can you just initialize this upfront?
```cpp
  static triton::common::HostPolicyCmdlineConfig empty_host_policy;
  const triton::common::HostPolicyCmdlineConfig* host_policy = &empty_host_policy;
  const auto policy_it = model->HostPolicyMap().find(host_policy_name);
  if (policy_it != model->HostPolicyMap().end()) {    
    host_policy = &policy_it->second;
  }
```

Reply:
Sure. Updated on commit: [Adjust host policy init logic](https://github.com/triton-inference-server/core/pull/207/commits/a21e40e101eea409b3e3cb522e011acb79f7b1e4)